src/main/java/school/lemon/changerequest/java/introduction/hw1/Task6.java

```diff
@@ -2,10 +2,19 @@
 
 public class Task6 {
     public static double calculateS(double x) {
-        return 0d;
+        return (1 + x + (Math.pow(x, 2)/factorial(2)) + (Math.pow(x, 3)/factorial(3)) + (Math.pow(x, 4)/factorial(4)));
+
+        }
+
+    private static int factorial(int n) {
+        for (int i = 1; i <= n; i++) {
+            n = n * i;
+            }
+            return n;
```

Comment:
Your factorial function doesn't return correct result.
Try to calculate on your own with n=2:
1. i=1; n=2; i>=n -> true; n=n*i; i++
2. i=2; n=2; i>=n -> true; n=n*i; i++
3. i=3; n=4; i>=n -> true; n=.....

As you see - you'll loop until you'll get overflow on int value and number will be negative.